doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -131,21 +138,65 @@ public void createDoiRequest(CreateDoiRequest createDoiRequest, String username)
         Publication publication = fetchPublicationForUser(createDoiRequest, username);
         verifyThatPublicationHasNoPreviousDoiRequest(publication);
         DoiRequest newDoiRequestEntry = createDoiRequestEntry(createDoiRequest, username);
-        publication.setDoiRequest(newDoiRequestEntry);
-        publication.setModifiedDate(newDoiRequestEntry.getModifiedDate());
+        updatePublication(publication, newDoiRequestEntry);
         putItem(publication);
     }
 
     @Override
-    public void updateDoiRequest(UUID publicationIdentifier, DoiRequestStatus requestedStatusChange,
+    public void updateDoiRequest(UUID publicationIdentifier, ApiUpdateDoiRequest apiUpdateDoiRequest,
```

Comment:
Not sure if ApiUpdateDoiRequest should leak into the service like this?

Reply:
I considered that, but then I would either have to send around two variables instead of one (status and message) and message could be null, so I would be sending around a null object and checking it here and there. The benefit is little at this point to send only the contents of the object and not the whole object itself since we need all the contents of the object